### PR TITLE
Memthol.0.4.0

### PR DESCRIPTION
**Pull request by hernoufM**
This PR :
- Discards warnings.
- Updates parser to support to the recent .ctf format generated by Memtrace.
- Updates inter-documentation links to the new Rust link's format.
- Crate dependencies was updated to the most recent version (except of some of them).
- Build script for `memthol`, that compiles `client` crate to wasm before compilation of himself. That was done to enable installation with ```cargo install```

### CODE REVIEW

**Review by krtab — APPROVED** [review]

**Comment by krtab** [comment]
Woops, I missclicked, only wanted to validate one commit.

**Comment by hernoufM** [comment]
No problems :smile:

File: libs/charts/src/filter/sub.rs [hunk 1]
```diff
@@ -21,7 +21,7 @@
 //!
 //! A sub filter is what [`Filter`]s are made of.
 //!
-//! [`Filter`]: ../struct.Filter.html (The Filter struct).
+// [`Filter`]: Filter (The Filter struct).
```
Comment (AdrienChampion) [inline libs/charts/src/filter/sub.rs:24]: Typo

**Comment by AdrienChampion** [comment]
- [ ] clean doc-links

Generic remark on doc-links: whenever `Item` is in scope where you're writing doc, you can just write `` [`Item`] `` to generate a link. This was not the case when I wrote this code, but now you can remove a lot of citation-style links like

```rust
/// [`SomeType`]: SomeType
```

**Comment by AdrienChampion** [comment]
It seems you removed `libs/client` from the workspace. The problem with doing this is that you cannot clean the project anymore without manually removing `libs/client/target`. I understand it's because we need to build the wasm part which takes a lock already taken by the server build (hence the use of `cargo-make` previously), but are you positive that's how other projects handle this issue?

File: libs/client/src/js.rs [hunk 2]
```diff
@@ -43,7 +43,7 @@ use wasm::*;
 
 #[wasm_bindgen]
 extern "C" {
-    /// Issues an alert.
+    // Issues an alert.
```
Comment (AdrienChampion) [inline libs/client/src/js.rs:46]: Typo?

**Review by AdrienChampion — CHANGES_REQUESTED** [review]
- [x] clean doc-links
- [x] rewrite all `1.0.0` bumps to `0.` as we're not ready for semantic versioning

**Comment by hernoufM** [comment]
> Generic remark on doc-links: whenever `Item` is in scope where you're writing doc, you can just write `` [`Item`] `` to generate a link. This was not the case when I wrote this code, but now you can remove a lot of citation-style links like
> 
> ```rust
> /// [`SomeType`]: SomeType
> ```

I hesitated to  remove the `[`SomeType`]: SomeType (Some explanation) ` notations because every of them was accompanied with the tip `Some explanation`, that appears when hoovers over link. But i guess, there are other ways to let those tips using new docs format.

**Comment by hernoufM** [comment]
> EDIT: couldn't the `build.rs` trigger on `cargo clean` and actually clean?

Actually, build.rs, is executed before compilation of memthol's binary crate, and that in turn needs compiled version of `client` at compile time. So, solution is, in my opinion, to let *build.rs* to move his generated artifacts into the root *target* directory, so it could be removed by `cargo clean`.

File: libs/alloc_data/Cargo.toml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "alloc_data"
-version = "0.3.0"
+version = "1.0.0"
```
[the remark this answers is not in the document]
Comment (hernoufM) [reply libs/alloc_data/Cargo.toml:3]: Yes, totally agree!

File: libs/alloc_data/src/lib.rs [hunk 4]
```diff
@@ -208,7 +208,7 @@ impl Builder {
         labels: Labels,
         toc: time::SinceStart,
         tod: Option<time::SinceStart>,
-    ) -> Self {
+    ) -> Self {
```
[the remark this answers is not in the document]
Comment (hernoufM) [reply libs/alloc_data/src/lib.rs:211]: Oups... I ll do that.

**Comment by AdrienChampion** [comment]
> > EDIT: couldn't the `build.rs` trigger on `cargo clean` and actually clean?
> 
> Actually, build.rs, is executed before compilation of memthol's binary crate, and that in turn needs compiled version of `client` at compile time. So, solution is, in my opinion, to let _build.rs_ to move his generated artifacts into the root _target_ directory, so it could be removed by `cargo clean`.

Sounds great 🐱

**Comment by AdrienChampion** [comment]
> > Generic remark on doc-links: whenever `Item` is in scope where you're writing doc, you can just write `` [`Item`] `` to generate a link. This was not the case when I wrote this code, but now you can remove a lot of citation-style links like
> > ```rust
> > /// [`SomeType`]: SomeType
> > ```
> 
> I hesitated to remove the `[`SomeType`]: SomeType (Some explanation) ` notations because every of them was accompanied with the tip `Some explanation`, that appears when hoovers over link. But i guess, there are other ways to let those tips using new docs format.

Yeah let's just use the normal workflow and let `rustdoc` handle tooltips and such. It's more concise and makes docs homogeneous compared to other crates.

**Review by AdrienChampion — APPROVED** [review]
LGTM

**Pull request merged**